app/controllers/metrics_controller.rb

```diff
@@ -33,12 +33,52 @@ def get_watchlist_instances
 		# On error, a 404 error is returned
 		begin
 			course_name = params[:course_name]
-			conditions = WatchlistInstance.get_instances_for_course(course_name)
-			render json: conditions
+			instances = WatchlistInstance.get_instances_for_course(course_name)
+			render json: instances
 		rescue => error
-			render :text => 'Not Found', :status => '404'
+			render json: {error:error.message}, status: :not_found
 			return
 		end
 	end
 
+	action_auth_level :update_watchlist_instances, :instructor
+	def update_watchlist_instances
+		# This API endpoint updates watchlist instances for a particular course
+		# On success, the watchlist instance will be updated appropriately
+		# params required would be the course name
+		# example json body {"method":"resolve","ids":[1,2,3]}
+		# method: update, resolve
+		# ids: [1,2,3...] list of ids to be updated
+		
+		begin
+			course_name = params[:course_name]
+		rescue => error
+			render json:  {error:"Course Not Found"}, :status => :not_found
+			return
+		end
```

Comment:
I believe when params does not contain a pair with the key being course_name, it will simply return nil. The double check needed here is probably to try to obtain a course row with the provided course name (raise an error if course_name is nil) and see if such a row exists and raise an error if it doesn't.

Reply:
okay! I will amend that accordingly